cl_dll/hud_spectator.cpp

```diff
@@ -1417,27 +1429,30 @@ bool CHudSpectator::ParseOverviewFile( )
 
 void CHudSpectator::LoadMapSprites()
 {
-	// right now only support for one map layer
-	if (m_OverviewData.layers > 0 )
+	for (auto& layer : m_OverviewData.layers)
 	{
-		m_MapSprite = gEngfuncs.LoadMapSprite( m_OverviewData.layersImages[0] );
+		if (layer.imagePath.empty())
+			layer.mapSprite = (struct model_s*) gEngfuncs.GetSpritePointer(m_hsprUnkownMap);
+		else
+			layer.mapSprite = gEngfuncs.LoadMapSprite(layer.imagePath.c_str());
 	}
-	else
-		m_MapSprite = NULL; // the standard "unkown map" sprite will be used instead
 }
 
 void CHudSpectator::DrawOverviewLayer()
 {
 	float screenaspect, xs, ys, xStep, yStep, x,y,z;
 	int ix,iy,i,xTiles,yTiles,frame;
 
-	qboolean	hasMapImage = m_MapSprite?TRUE:FALSE;
-	model_t *   dummySprite = (struct model_s *)gEngfuncs.GetSpritePointer( m_hsprUnkownMap);
+	Vector camOrigin, camAngles;
+	GetCameraView(camOrigin, camAngles);
+
+	const auto currLayer = GetCurrentLayer(camOrigin);
+	const auto hasMapImage = currLayer.mapSprite == (struct model_s*)gEngfuncs.GetSpritePointer(m_hsprUnkownMap) ? false : true;
 
 	if ( hasMapImage)
 	{
-		i = m_MapSprite->numframes / (4*3);
-		i = sqrt((float)i);
+		i = currLayer.mapSprite->numframes / (4*3);
+		i = (int) sqrt((float)i);
```

Comment:
C++ implicitly converts float to int.
```suggestion
		i = sqrt((float)i);
```